src/Concordium/Client/Types/Contract/Schema.hs

```diff
@@ -449,6 +509,9 @@ instance AE.ToJSON SchemaType where
     ILeb128 _ -> AE.String "<String with signed integer>"
     ByteList _ -> AE.String "<String with lowercase hex>"
     ByteArray _ -> AE.String "<String with lowercase hex>"
+    TaggedEnum variants -> AE.object ["TaggedEnum" .=
+      (toJsonArray . map (\(k, v) -> AE.object [AE.fromText k .= v]) $
+      (\(k,v) -> (pack $ show k, v)) <$> Map.toList variants)]
```

Comment:
Can you please write this in a more readable way. Parsing this is terribly complicated now. Name the intermediate values at least.
What is the intended display here even? I would have thought the tags should be ignored, since they are not relevant for what the user needs to write.

Reply:
The intention is to display the TaggedEnum as JSON. I suppose whether the tag should be displayed depends on the semantics and how the client is meant to be used. 

If the string is a key I guess it does not have to, if not, I think it better be present. Tags are also displayed in the output of concordium-cargo to the my best knowledge.

Reply:
In what context are the tags printed?

As the comment above states, the output here should be roughly what the user is expected to write to supply the value.

As far as I see in the cargo-concordium code the user is meant to supply the same format as in the `Enum` case, so no tags.

Can you please make sure that this agrees?

Reply:
I haven't tested cargo-concordium with the event schemas, but it should be sufficient to provide the variant name and then use that to find the index in the schema.

Reply:
I'll have a look at cargo-concordium and make it is consistent with how it's done there.